File: samples/Clojure/99-bottles-of-beer.clj
(defn verse
  [n]
  (printf "%d bottles of beer on the wall,
%d bottles of beer,
Take one down, pass it around,
%d bottles of beer on the wall.\n\n"
          n
          n
          (dec n)))

(defn sing
  [start]
  (doseq [n (range start 0 -1)]
    (verse n)))

(sing 99)

File: samples/Objective-C/99-bottles-of-beer.m
#import <Foundation/Foundation.h>

int main()
{
    @autoreleasepool {
        int bottles = 99;
        do
        {
            NSLog(@"%i bottles of beer on the wall\n", bottles);
            NSLog(@"%i bottles of beer\n", bottles);
            NSLog(@"Take one down, pass it around\n");
            NSLog(@"%i bottles of beer on the wall\n\n", --bottles);
        } while (bottles > 0);
	
    }
    return 0;
}
